### DISCUSSION TITLE
Parsing the data after being processed by functions

### DISCUSSION THREAD

**Discussion by tansuozhe123** [lf-edge/ekuiper#2471] [Q&A]
The result of the source data processed by the function will wrap a layer, 
for example：
Raw data:{"key":"value"}
select parse(*) from mqtt; 
The processed data {parse:{"key":"value1"}}, how to remove the new layer?

**Comment by ngjaying** [comment]
Use sink property dataField, set dataField=parse

**Comment by tansuozhe123** [comment]
Can we fetch the next level of data?
For example:
Data :{message:{"value":{"k":"v"}}} To get the innermost {"k":"v"}, how is the dataField parameter set,dataField =message.value?

**Reply by ngjaying** [reply]
You can use dataTemplate for complex scenario.

**Comment by tansuozhe123** [comment]
Okay, that's it. Thank you very much
